src/team.coffee

```diff
@@ -22,196 +22,156 @@
-      message += countMessage if countMessage
-      msg.send message
-
-  teamExists = (team_name, msg) ->
-    return true if team_name is config.default_team_label
-
-    if robot.brain.data.teams[team_name]
-      true
-    else
-      msg.send "#{teamLabel(team_name)} does not exist"
-      false
+  unless Config.adminList()
+    robot.logger.warning 'HUBOT_TEAM_ADMIN environment variable not set'
 
   ##
   ## hubot create <team_name> team - create team called <team_name>
   ##
   robot.respond /create (\S*) team ?.*/i, (msg) ->
-    team_name = msg.match[1]
-    addTeam(team_name, msg)
+    teamName = msg.match[1]
+    message = if team = Team.create teamName
+                responseMessage.teamCreated(team)
+              else
+                responseMessage.teamAlreadyExists(teamName)
+    msg.send message
 
   ##
   ## hubot (delete|remove) <team_name> team - delete team called <team_name>
   ##
   robot.respond /(delete|remove) (\S*) team ?.*/i, (msg) ->
-    team_name = msg.match[2]
-    return unless teamExists(team_name, msg)
+    teamName = msg.match[2]
+    return msg.reply responseMessage.adminRequired() unless msg.message.user.name in Config.admins()
+    message = if Team.destroy(teamName)
+                responseMessage.teamDeleted(teamName)
+              else
+                responseMessage.teamNotFound(teamName)
+    msg.send message
 
-    if msg.message.user.name in admins
-      unless team_name is config.default_team_label
-        delete robot.brain.data.teams[team_name]
-        msg.send "Team `#{team_name}` removed"
-    else
-      msg.reply "Sorry, only admins can remove teams"
 
   ##
   ## hubot list teams - list all existing teams
   ##
   robot.respond /list teams ?.*/i, (msg) ->
-    team_count = Object.keys(robot.brain.data.teams).length
-    team_count = team_count - 1 if robot.brain.data.teams[config.default_team_label]
-
-    if team_count > 0
-      message = "Teams:\n"
-
-      for team of robot.brain.data.teams
-        continue if team is config.default_team_label
-        size = teamSize(team)
-        if size > 0
-          message += "`#{team}` (#{size} total)\n"
-          for user in robot.brain.data.teams[team]
-            message += "- #{user}\n"
-          message += "\n"
-        else
-          message += "`#{team}` (empty)\n"
-
-    else
-      message = "No team was created so far"
-
-    msg.send message
+    teams = Team.all()
+    msg.send responseMessage.listTeams(teams)
 
   ##
   ## hubot <team_name> team add (me|<user>) - add me or <user> to team
   ##
   robot.respond /(\S*)? team add (\S*) ?.*/i, (msg) ->
-    team_name = msg.match[1]
-    user = msg.match[2]
-    if user.toLocaleLowerCase() == "me"
-      user = msg.message.user.name
-    addUserToTeam(user, team_name, msg)
+    teamName  = msg.match[1]
+    team      = if teamName
+                  Team.find(teamName)
+                else
+                  Team.default()
```

Comment:
This can be written in one line:

```js
team = if teamName then Team.find(teamName) else Team.default()
```

And don't align assignments, only one white space between operands.